Assets/Scripts/ObjectScripts/PlayerHop.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using UnityEngine;
+
+public class PlayerHop : MonoBehaviour
+{
+    private Rigidbody _rb;
+    private Vector3 _startPos;
+    private bool _hopping;
+    private double _marginOfError;
+
+    private void Start()
+    {
+        _rb = GetComponent<Rigidbody>();
+        _startPos = transform.position;
+        _hopping = false;
+        _marginOfError = MusicPlayer.current.marginOfError;
+    }
+
+    private void Update()
+    {
+        if (transform.position.y < _startPos.y)
+        {
+            transform.position = _startPos;
+            _hopping = false;
+        }
+        if (Math.Abs(MusicPlayer.current.GetAudioSourceTime() - PlayerAction.Current.GetNextTimestamp()) < _marginOfError)
```

Comment:
Is the _marginOfError needed for this? Correct me if I'm wrong, but this is mainly used for player to get perfect even they are a bit off, I don't think decoration jump needs it.

Reply:
You're right. I was having issues where if it was lower than _marginOfError (like 0.1 instead of 0.3), the decorations didn't hop and I wasn't sure why. Turns out the original implementation was based on _inputIndex, so if the player jumped within 0.25f, it would increase _inputIndex, and the decorations wouldn't hop. Fixed it by using a separate _hopIndex.

I switched _marginOfError to 0.1f. I tried 0.05f, but since we're checking in Update() (which is fps dependent), I think there might be times where the check doesn't happen quickly enough to detect that it is within the interval. But let me know if you think it should be lower; we can give it a shot

Reply:
That looks great for me. Visually you can't really tell there is delay so I think that's more than enough